### PR TITLE
Add the demo run to the website pages

### LINKED ISSUE #426: Add a demo run to the website pages
**Is your feature request related to a problem? Please describe.**

We now have a working system and a draft demonstration of the run process, which we could add as a website page.

**Describe the solution you'd like**

Add the demo to the `sphinx` source.

### PR DESCRIPTION
**Pull request by davidorme**
# Description

This adds a new page to the docs that contains the current simple walkthrough of running the model. It fixes a couple of minor `sphinx` issues along the way.

See: https://virtual-ecosystem.readthedocs.io/en/426-add-a-demo-run-to-the-website-pages/virtual_ecosystem/virtual_ecosystem_in_use.html

Closes #426 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_ecosystem/example_data/config/data_config.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [core.data_output_options]
-save_initial_state = false
+save_initial_state = true
```
Comment (jacobcook1995) [inline virtual_ecosystem/example_data/config/data_config.toml:2]: Is this needed to get the display to work on the website?

**Review by jacobcook1995 — APPROVED** [review]
LGTM! Only comment is that it isn't as engrossing as the live demo as it doesn't run while you watch it, but I'm guessing that isn't straightforward to setup the docs to do.

File: virtual_ecosystem/example_data/config/data_config.toml (hunk 1 above)
Comment (davidorme) [reply virtual_ecosystem/example_data/config/data_config.toml:2]: It's needed to generate those plots from the initial data! We could just point back to the original files, but it seems worth demonstrating that step.

**Comment by davidorme** [comment]
Yeah - there's no way to generate this dynamically on request. We'd need a live `jupyter` server running an instance - which we could do but probably not now 😄

**Comment by jacobcook1995** [comment]
Yeah that definitely feels like overkill for this PR

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+---
+
+# The Virtual Ecosystem model
+
+The code below is a brief demonstration of the Virtual Ecosystem model in operation.
+The workflow of the model is:
+
+## Create the model configuration and initial data
+
+Here we are using the example data files supplied with the `virtual_ecosystem`
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:24]: ``
```suggestion
Here we are using the example data supplied with the `virtual_ecosystem`
```

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+fi
+```
+
+```{code-cell}
+%%bash
+# Install the example data directory from the Virtual Ecosystem package
+ve_run --install-example /tmp/
+```
+
+The `ve_example` directory contains the following files. These include:
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:54]:
```suggestion
The `ve_example` directory contains the following files:
```

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+These files are written to the standard NetCDF data file format.
+
+```{code-cell}
+# Load the generated data files
+initial_state = xarray.load_dataset("/tmp/ve_example/out/initial_state.nc")
+continuous_data = xarray.load_dataset("/tmp/ve_example/out/all_continuous_data.nc")
+final_state = xarray.load_dataset("/tmp/ve_example/out/final_state.nc")
+```
+
+### Initial state
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:121]:
```suggestion
### Initial state and input data
```
You show the climate input time series here so I think we should include this in the title

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+The `initial_state.nc` file contains all of the data required to start up the model.
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:123]: Add something like: It also contains time series of climate variables at reference height that drive the model.

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+poly = ax.bar3d(x, y, bottom, width, depth, top, shade=True, color=colors)
+ax.set_title("Elevation (m)")
+
+cell_bounds = range(0, 811, 90)
+ax.set_xticks(cell_bounds)
+ax.set_yticks(cell_bounds);
+```
+
+For other variables, such as air temperature and precipitation, the initial data
+also provides time series data across the grid cells that are used to force the
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:178]:
```suggestion
also provides time series data at reference height that are used to force the
```

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+initial_state
+```
+
+```{code-cell}
+# Make two side by side plots
+fig, (ax1, ax2) = plt.subplots(ncols=2, figsize=(12, 5))
+
+# Air temperature
+ax1.plot(initial_state["time_index"], initial_state["air_temperature_ref"])
+ax1.set_title("Air temperature across grid cells")
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:191]:
```suggestion
ax1.set_title("Air temperature at reference height")
```

File: docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+ax1.set_ylabel("Air temperature (°C)")
+ax1.set_xlabel("Time step (months)")
+
+# Precipitation
+ax2.plot(initial_state["time_index"], initial_state["precipitation"])
+ax2.set_title("Precipitation across grid cells")
```
Comment (vgro) [inline docs/source/virtual_ecosystem/virtual_ecosystem_in_use.md:197]:
```suggestion
ax2.set_title("Precipitation forcing")
```

**Review by vgro — APPROVED** [review]
Looks good to me, a few small comments regarding the input data.

**Comment by davidorme** [comment]
> Yeah that definitely feels like overkill for this PR

I think it's probably as simple as using Binder? https://mybinder.org/

**Pull request merged**